packages/ckeditor5-watchdog/src/editorwatchdog.ts

```diff
@@ -284,13 +295,31 @@ export default class EditorWatchdog<TEditor extends Editor = Editor> extends Wat
 	}
 
 	/**
-	 * Returns the editor data.
+	 * Gets the all data that are required to reinitialize editor instance.
 	 */
-	private _getData(): Record<string, string> {
-		const data: Record<string, string> = {};
+	private _getData(): EditorData {
+		const editor = this.editor!;
+		const rootNames = editor.model.document.getRootNames();
+		const data: EditorData = {
+			roots: {},
+			markers: {}
+		};
+
+		rootNames.forEach( rootName => {
+			const root = editor.model.document.getRoot( rootName )!;
+
+			data.roots[ rootName ] = {
+				content: JSON.stringify( Array.from( root.getChildren() ) ),
+				attributes: JSON.stringify( Array.from( root.getAttributes() ) )
+			};
+		} );
 
-		for ( const rootName of this._editor!.model.document.getRootNames() ) {
-			data[ rootName ] = this._editor!.data.get( { rootName } );
+		for ( const marker of editor.model.markers ) {
+			data.markers[ marker.name ] = {
```

Comment:
Filter out markers that do not affect the data (we talked about it for the "next step" but AFAIU it makes sense here as well).